libsplinter/src/error/constraint_violation.rs

```diff
@@ -0,0 +1,220 @@
+// Copyright 2018-2020 Cargill Incorporated
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//! Module containing ConstraintViolationError implementation.
+
+use std::error;
+use std::fmt;
+
+struct Source {
+    source: Box<dyn error::Error>,
+}
```

Comment:
Why is this a struct instead of just using the `Box<dyn error::Error>` directly in the `ConstraintViolationError`?